src/Symfony/Component/EventDispatcher/DependencyInjection/RegisterListenersPass.php

```diff
@@ -183,6 +171,54 @@ private function getEventFromTypeDeclaration(ContainerBuilder $container, string
 
         return $name;
     }
+
+    /**
+     * Resolve the listener method:
+     *   - look for a method named after the event (camelized, with "on" prefix)
+     *   - if such a method does not exist, check if only one public method exist which expect the event as parameter
+     *   - otherwise, throw an exception because we need more information.
+     */
+    private function resolveListenerMethodFromTypeDeclaration(ContainerBuilder $container, string $id, string $eventName, array $aliases): string
+    {
+        $method = 'on'.preg_replace_callback([
+            '/(?<=\b|_)[a-z]/i',
+            '/[^a-z0-9]/i',
+        ], fn ($matches) => strtoupper($matches[0]), $eventName);
+        $method = preg_replace('/[^a-z0-9]/i', '', $method);
+
+        if (
+            null === ($class = $container->getDefinition($id)->getClass())
+            || !($r = $container->getReflectionClass($class, false))
+            || $r->hasMethod($method)
+        ) {
+            return $method;
+        }
+
+        $publicMethods = $r->getMethods(\ReflectionMethod::IS_PUBLIC);
+        $eventName = array_flip($aliases)[$eventName] ?? $eventName;
+
+        $candidateMethods = [];
+        foreach ($publicMethods as $publicMethod) {
+            if (
+                !$publicMethod->isStatic()
+                && 1 === $publicMethod->getNumberOfRequiredParameters()
```

Comment:
This check is bad given that the EventDispatcher component actually provides 3 parameters to listeners:
- the event object
- the event name
- the EventDispatcher instance